crypto/evp/pmeth_lib.c

```diff
@@ -247,10 +247,11 @@ static EVP_PKEY_CTX *int_ctx_new(OSSL_LIB_CTX *libctx,
      */
     if (e != NULL)
         pmeth = ENGINE_get_pkey_meth(e, id);
-    else if (pkey != NULL && pkey->foreign)
+    else
+# endif /* OPENSSL_NO_ENGINE */
+        if (pkey != NULL && pkey->foreign)
         pmeth = EVP_PKEY_meth_find(id);
```

Comment:
Indentation error here

Reply:
No, this is on purpose, because it completes the `else if` started in line 250.

Reply:
I know this looks weird, but we have tons of incomplete else clauses of that kind in our source code.

Reply:
Just try

    git grep -n -B 1 '^# *\(else\|endif\)' | grep -A 1 '^[^#]*else'

Reply:
Its not the "if" itself that is incorrect - I understand that is why it looks like that. It's line 253 which needs additional indentation

Reply:
Actually, thinking about this more I would line the if up with the "else"

Reply:
Your example is not the same....its not an "else if" construct.

Reply:
If you have a better suggestion how to get the indentation right both with and without the guard (without duplicating code), and  how to draw attention to those dangerous incomplete else-clauses wich can easily be overlooked and cause `no-xyz` failures, then I'll be happy to address this problem in a separate pull request.

Reply:
Ok, our last replies crossed. I will follow your suggestion Matt.

Reply:
Last [force-push](https://github.com/openssl/openssl/compare/0c014974cd2fd636cc58ec500ea75e0611a81a43..6de9e91c0c9175f95a6f43c7d141b0892906a7d0) amends my commit.